src/constructs/aws/DatabaseSql.ts

```diff
@@ -0,0 +1,161 @@
+import { CfnOutput, Duration, SecretValue } from "@aws-cdk/core";
+import type { FromSchema } from "json-schema-to-ts";
+import {
+    Credentials,
+    DatabaseInstance,
+    DatabaseInstanceEngine,
+    MariaDbEngineVersion,
+    MysqlEngineVersion,
+    PostgresEngineVersion,
+} from "@aws-cdk/aws-rds";
+import { InstanceType, SubnetType } from "@aws-cdk/aws-ec2";
+import type { IInstanceEngine } from "@aws-cdk/aws-rds/lib/instance-engine";
+import { AwsConstruct } from "@lift/constructs/abstracts";
+import type { AwsProvider } from "@lift/providers";
+
+const SCHEMA = {
+    type: "object",
+    properties: {
+        name: {
+            type: "string",
+            pattern: "^[\\w\\d-_]*$/",
+        },
+        engine: {
+            type: "string",
+            enum: ["mysql", "mariadb", "postgres"],
+        },
+        username: {
+            type: "string",
+        },
+        password: {
+            type: "string",
+            minLength: 8,
+        },
+        instanceType: { type: "string" },
+        storageSize: {
+            type: "integer",
+            minimum: 20,
+        },
+    },
+    additionalProperties: false,
+    required: [],
+} as const;
+
+type Configuration = FromSchema<typeof SCHEMA>;
+
+export class DatabaseSql extends AwsConstruct {
+    public static type = "database/sql";
+    public static schema = SCHEMA;
+
+    private readonly dbInstance: DatabaseInstance;
+    private readonly dbHostOutput: CfnOutput;
+    private readonly username: string;
+    private passwordSecretName: string | undefined;
+
+    constructor(
+        scope: CdkConstruct,
+        private readonly id: string,
+        private readonly configuration: Configuration,
+        private readonly provider: AwsProvider
+    ) {
+        super(scope, id);
+
+        const vpc = provider.enableVpc();
+
+        this.username = this.configuration.username ?? "admin";
+
+        this.dbInstance = new DatabaseInstance(this, "Instance", {
+            // https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseInstance.html#construct-props
+            instanceIdentifier: configuration.name ?? `${this.provider.stackName}-${id}`,
+            databaseName: this.safeDbName(configuration.name ?? `${this.provider.stackName}-${id}`),
+            engine: this.getEngineVersion(),
+            instanceType: new InstanceType(configuration.instanceType ?? "t3.micro"),
+            credentials: this.credentials(),
+            vpc,
+            // Put the instance in the private subnet
+            vpcSubnets: {
+                subnetType: SubnetType.PRIVATE,
+            },
+            // We go with 20 (the minimum) instead of 100 by default, because it is much cheaper
+            allocatedStorage: configuration.storageSize ?? 20,
+            // 7 days backups instead of the default 1
+            backupRetention: Duration.days(7),
+            cloudwatchLogsExports: this.logsToEnable(),
+        });
+
```

Comment:
Maybe we need to add a security group to allow inbound from the lambda security group?